src/devices/src/virtio/block/device.rs

```diff
@@ -320,16 +321,19 @@ impl Block {
                     head.index, e
                 )
             });
+
+            if queue.prepare_kick(mem) {
+                self.irq_trigger
+                    .trigger_irq(IrqType::Vring)
+                    .unwrap_or_else(|_| {
+                        METRICS.block.event_fails.inc();
+                    });
+            }
```

Comment:
This introduces a regression for cases where notif suppression is not enabled/negotiated: with this new code, IRQs are being generated _on each descriptor_ instead of just one IRQ after the full chunk of currently avail descriptors.

I am expecting this regression to be actually visible in perf testing with notif suppres disabled/unavailable in the guest.

Reply:
I believe we can structure the code so as to avoid regressions on the _legacy_ path.

Reply:
I was unsure if we should treat this case. It complicates the code a bit, and notification suppression is present in all the kernels that we support AFAIK. Anyway I tried to fix this. I added a new commit. PTAL.